src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -1,175 +1,506 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Diagnostics;
+using static System.Math;
 
 namespace Roslyn.Utilities
 {
-    internal static class EditDistance
+    internal class EditDistance : IDisposable
     {
-        private const int MaxPooledArraySize = 256;
+        private string originalText;
+        private char[] originalTextArray;
+        private int threshold;
 
-        // Keep around a few arrays of size 256 that we can use for operations without
-        // causing lots of garbage to be created.  If we do compare items larger than
-        // that, then we will just allocate and release those arrays on demand.
-        private static ObjectPool<int[]> s_pool = new ObjectPool<int[]>(() => new int[MaxPooledArraySize]);
+        // Cache the result of the last call to IsCloseMatch.  We'll often be called with the same
```

Comment:
And do we want perf tests to make sure this mechanism is sufficient (/stays sufficient)?